src/sinks/util/retries.rs

```diff
@@ -193,12 +193,14 @@ where
                     );
                     Some(self.build_retry())
                 } else {
+                    // For now, retry request in all cases
```

Comment:
this is nice. 
it took me sometime to understand. can you add a bit more explanation referencing line 177/189.
basically what is happening at line 177/189 and so when we would end up here.